static/src/javascripts/projects/common/modules/identity/account-profile.js

```diff
@@ -96,8 +104,58 @@ define([
         }
     };
 
+    accountProfile.prototype.avatarUploadByGAE = function (avatarForm) {
+        var self = this;
+        var xhr = self.createCORSRequest('GET', self.urls.avatarTokenUrl);
+
+        if (!xhr) {
+            self.prependErrorMessage(self.messages.noCorsError, avatarForm);
+        }
+
+        xhr.onerror = function () {
+            self.prependErrorMessage(self.messages.noServerError, avatarForm);
+        };
+
+        xhr.onload = function () {
+            avatarForm.setAttribute('action', xhr.responseText);
+            avatarForm.submit();
+        };
+
+        xhr.send();
+    };
+
+    accountProfile.prototype.avatarUploadByApi = function (avatarForm) {
+        var self = this;
```

Comment:
Yeah, ignore me, just did a search of the rest of the codebase and it's used all over the place so it's probably not worth worrying about. If you wanted to, you'd probably need to do things like this:

```
xhr.onerror = function () {
      // this is now bound to  the outer scope
      self.prependErrorMessage(self.messages.noServerError, avatarForm);
}.bind(this);
```
But you'd need to apply that consistently, so ignore me for this PR.